zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java

```diff
@@ -0,0 +1,188 @@
+   */
+  public Predictions predict(final List<?> data) throws IOException, MlEnginePredictException {
+    final GoogleCloudMlV1PredictRequest predict =
+        new GoogleCloudMlV1PredictRequest().set("instances", data);
+    final GoogleApiHttpBody httpBody = instance().projects()
+        .predict(id().value(), predict)
+        .execute();
+
+    final Response response = Response.from(httpBody);
+    return response.predictions().orElseThrow(() -> {
+      return response.error().map(MlEnginePredictException::new).get();
+    });
+  }
+
+  /**
+   * Predict.
+   *
+   * @param examples TensorFlow {@link Example} input data.
+   */
+  public Predictions predictExamples(final List<Example> examples)
+      throws IOException, MlEnginePredictException {
+    final List<Map<String, String>> data = examples.stream().map(example -> {
+      final byte[] bytes = example.toByteArray();
+      final String b64 = BaseEncoding.base64().encode(bytes);
+
+      return Collections.singletonMap("b64", b64);
+    }).collect(Collectors.toList());
+
+    return predict(data);
+  }
+
+  /**
+   * Close the model.
+   */
+  @Override
+  public void close() throws Exception {
+    httpTransport().shutdown();
+  }
+
+  /**
+   * Prediction response.
+   */
+  @AutoValue
+  public abstract static class Response {
+
+    abstract GoogleApiHttpBody content();
+
+    static Response from(final GoogleApiHttpBody content) {
+      return new AutoValue_MlEngineModel_Response(content);
+    }
+
+    /**
+     * List of predictions. Return type depends on the model used.
+     *
+     * @see <a href="https://cloud.google.com/ml-engine/docs/v1/predict-request">https://cloud.google.com/ml-engine/docs/v1/predict-request</a>
+     */
+    public Optional<Predictions> predictions() {
+      final List<Object> predictions = (List<Object>) content()
+          .getOrDefault("predictions", Collections.emptyList());
+
+      if (predictions.isEmpty()) {
+        return Optional.empty();
+      }
+
+      return Optional.of(Predictions.create(predictions));
+    }
+
+    /** Prediction error. */
+    public Optional<String> error() {
+      return Optional.ofNullable((String) content().get("error"));
+    }
+
+    /** Holds the predictions values. */
+    @AutoValue
+    public abstract static class Predictions {
+
+      private static final ObjectMapper MAPPER = new ObjectMapper();
+
+      /**
+       * List of predictions. Return type depends on the model used.
+       *
+       * @see <a href="https://cloud.google.com/ml-engine/docs/v1/predict-request">https://cloud.google.com/ml-engine/docs/v1/predict-request</a>
+       */
+      public abstract List<Object> values();
+
+      /**
+       * List of predictions.
+       *
+       * @param klass class to each returned prediction objects are converted.
+       */
+      public <T> List<T> values(final Class<T> klass) {
+        return values()
+            .stream()
+            .map(p -> MAPPER.convertValue(p, klass))
```

Comment:
Is some kind of memorization of `MAPPER` for specific `klass` needed here?

Reply:
good point, but i'm not sure if can do it, let me have a look.